### PR TITLE
feat(app): integrate the block-sdk

### PR DESCRIPTION
**Pull request by g3ndo9kar9**
### In This PR 
- I integrate the block-sdk
    - I integrate the mev, default lanes (ordered accordingly) into the `LaneMempool` which will be used as the app's `Mempool`
    - I integrate the Prepare / ProcessProposal handlers and custom CheckTx handlers
         - Prepare / ProcessProposal will be responsible for building blocks
         - The CheckTx handler is responsible for
             - Performing extra verification on bid-txs
             - Purging txs from cometBFT's mempool if they aren't in the app
    - Modules
        - I integrate the `x/auction` module, and the corresponding `AnteHandler` for the auction module
### Testing
- We've tested this integration against our own set of load-tests. These tests, spin up a network of 5 nodes (3 val, 1 seed, 1 full). Perform various perturbations against them (killing, restarting, disconnecting from the network, etc.), and periodically send a load of 350 tx/sec for 90s. The output from one of our trials is [here](https://snapshots.raintank.io/dashboard/snapshot/y23ErIIGdQEL8AXCbFdv4grb2hxldbMz?orgId=2), happy to go through these in more depth / review the methodology.
### Let's Discuss
- I configured the x/auction module in the `v202` upgrade handler. The params chosen are [here](https://github.com/skip-mev/neutron/pull/2/files#diff-7c95504b0c554fdd10576f31ea32ca63a0321c2ddf0bb85f79821c3257b0b338R22), happy to change these if requested
- I also configured the lane mempool capacities / other parameters [here](https://github.com/skip-mev/neutron/pull/2/files#diff-802c8455e6cfe3da0a5fb5d9ff1f32ca2e651f0a0434163eb700887a0c871391R11)

### CODE REVIEW

File: app/ante_handler.go [hunk 1]
```diff
@@ -77,9 +88,14 @@ func NewAnteHandler(options HandlerOptions, logger log.Logger) (sdk.AnteHandler,
 		ante.NewSetPubKeyDecorator(options.AccountKeeper),
 		ante.NewValidateSigCountDecorator(options.AccountKeeper),
 		ante.NewSigGasConsumeDecorator(options.AccountKeeper, sigGasConsumer),
-		ante.NewSigVerificationDecorator(options.AccountKeeper, options.SignModeHandler),
+		// ante.NewSigVerificationDecorator(options.AccountKeeper, options.SignModeHandler),
```
Comment (aljo242) [inline app/ante_handler.go]: Why are we removing this?

File: app/app.go [hunk 2]
```diff
@@ -928,6 +966,19 @@ func New(
...
+
+	mevLane, baseLane := app.CreateLanes()
+	mempool, err := blocksdk.NewLanedMempool(app.Logger(), []blocksdk.Lane{mevLane, baseLane})
+	if err != nil {
+		panic(err)
+	}
+
+	// set the mempool first
+	app.SetMempool(mempool)
+	app.Mempool = mempool
```
Comment (aljo242) [inline app/app.go]: What is the difference between these two lines?

File: app/app.go [hunk 3]
```diff
@@ -1033,6 +1123,28 @@ func (app *App) setupUpgradeHandlers() {
...
+// ChainID gets chainID from private fields of BaseApp
+// Should be removed once SDK 0.50.x will be adopted
+func (app *App) ChainID() string {
+	field := reflect.ValueOf(app.BaseApp).Elem().FieldByName("chainID")
+	return field.String()
+}
+
+// CheckTx will check the transaction with the provided checkTxHandler. We override the default
+// handler so that we can verify bid transactions before they are inserted into the mempool.
+// With the POB CheckTx, we can verify the bid transaction and all of the bundled transactions
```
Comment (aljo242) [inline app/app.go]:
```suggestion
// With the Block-SDK CheckTx, we can verify the bid transaction and all of the bundled transactions
```

File: app/ante_handler.go (hunk 1 above)
Comment (g3ndo9kar9) [reply app/ante_handler.go]: That was leftover from loadtesting

**Review by aljo242 — APPROVED** [review]
lgtm

File: app/upgrades/v2.0.2/upgrades.go [hunk 4]
```diff
@@ -5,20 +5,49 @@ import (
...
 	}
 }
+
+func setAuctionParams(ctx sdk.Context, feeBurnerKeeper *feeburnerkeeper.Keeper, auctionKeeper auctionkeeper.Keeper) error {
+	treasury := feeBurnerKeeper.GetParams(ctx).TreasuryAddress
+	_, data, err := bech32.DecodeAndConvert(treasury)
+	if err != nil {
+		return err
+	}
+	fmt.Println("treasury addy", feeBurnerKeeper.GetParams(ctx).TreasuryAddress)
```
Comment (Eric-Warehime) [inline app/upgrades/v2.0.2/upgrades.go]: Either log this or delete.

**Review by Eric-Warehime — APPROVED** [review]

File: app/upgrades/v2.0.2/upgrades.go (hunk 4 above)
Comment (davidterpay) [inline app/upgrades/v2.0.2/upgrades.go]:
```suggestion
	fmt.Println("treasury address", feeBurnerKeeper.GetParams(ctx).TreasuryAddress)
```
lmao u troll

**Review by davidterpay — APPROVED** [review]

**Pull request merged**